app/Models/DemandRequest.php

```
@@ -1,72 +1,268 @@
+            throw new RuntimeException("Cannot allocate resource to unfunded demand request.");
+        }
+
+        $this->update([
+            'allocated_resource_id' => $resource->id,
+            'status' => DemandStatus::Fulfilled,
+            'filled_at' => Carbon::now(),
+        ]);
     }
 
-    public function domain(): BelongsTo
+    public function reject(string $reason): void
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$reason'. (undefined)

(UnusedFormalParameter)